hoomd/mpcd/RadialSolventVelocityAnalyzer.cc

```diff
@@ -0,0 +1,201 @@
+// Copyright (c) 2009-2019 The Regents of the University of Michigan
+// This file is part of the HOOMD-blue project, released under the BSD 3-Clause License.
+
+// Maintainer: mphoward
+
+/*!
+ * \file mpcd/RadialSolventVelocityAnalyzer.cc
+ * \brief Declaration of mpcd::RadialSolventVelocityAnalyzer
+ */
+
+#include "RadialSolventVelocityAnalyzer.h"
+#include "hoomd/extern/pybind/include/pybind11/stl.h"
+
+namespace mpcd
+{
+
+/*!
+ * \param sysdata MPCD system data
+ * \param bin_width Width for binning particle distances
+ */
+RadialSolventVelocityAnalyzer::RadialSolventVelocityAnalyzer(std::shared_ptr<mpcd::SystemData> sysdata,
+                                                             Scalar bin_width)
+    : m_mpcd_sys(sysdata),
+      Analyzer(m_mpcd_sys->getSystemDefinition()),
+      m_mpcd_pdata(m_mpcd_sys->getParticleData()),
+      m_sysdef(m_mpcd_sys->getSystemDefinition()),
+      m_global_box(m_sysdef->getParticleData()->getGlobalBox()),
+      m_bin_width(bin_width),
+      m_Rmax(0.0),
+      m_num_samples(0)
+    {
+    m_exec_conf->msg->notice(5) << "Constructing RadialSolventVelocityAnalyzer" << std::endl;
+
+    assert(m_bin_width > 0.0);
+
+    // allocate memory for the bins
+    Scalar3 L = m_global_box.getL();
+    Scalar m_Rmax = std::min( L.x, std::min( L.y, L.z));
+    m_num_bins = std::ceil(m_Rmax / m_bin_width);
+    GPUArray<unsigned int> counts(m_num_bins, m_exec_conf);
+    m_counts.swap(counts);
+    GPUArray<double> accum_rho(m_num_samples, m_num_bins, m_exec_conf);
+    m_accum_rho.swap(accum_rho);
```

Comment:
Since this is only meant to work on the CPU, and we also don't actually know how many samples you're going to get, I recommend that you just use `std::vector` to save all of these. I'll comment on how to do that below.
```suggestion
```